docs/pipeline/triggers/COST_TRIGGER.md

```diff
@@ -0,0 +1,177 @@
+Rather than using this regret units directly, we build an area-under-the-curve metric.
+The area under the regret curve measures the time regret units have spent being unaddressed.
+
+As this policy operates the two metrics `time` (cost) and a regret metric we need
+a way to express the tradeoff between the two. A user e.g. has to specify how many seconds of training time he is
+willing to eradicate a certain amount of cumulative regret.
+
+### Main Architecture
+
+```mermaid
+classDiagram
+    class Trigger {
+        <<abstract>>
+        +void init_trigger(trigger_context)
+        +Generator[Triggers] inform(new_data)*
+        +void inform_new_model(model_id, ...)
+    }
+
+    class BatchedTrigger {
+        <<abstract>>
+        +void inform(...)
+        -bool evaluate_batch(batch, trigger_index)*
+    }
+
+    class PerformanceTriggerMixin {
+        +DataDensityTracker data_density
+        +PerformanceTracker performance_tracker
+        +StatefulModel model
+
+        -EvaluationResults run_evaluation(interval_data)
+    }
+
+    class CostTrigger {
+        <<abstract>>
+        +CostTracker cost_tracker
+        +IncorporationLatencyTracker latency_tracker
+
+        +void init_trigger(trigger_context)
+        +void inform_new_model(model_id, ...)
+
+        -bool evaluate_batch(batch, trigger_index)
+        -float compute_regret_metric()*
+
+    }
+
+    class CostTracker {
+    }
+
+    class IncorporationLatencyTracker {
+    }
+
+    class DataIncorporationLatencyCostTrigger {
+        -float compute_regret_metric()
+    }
+
+    class AvoidableMisclassificationCostTrigger {
+        NumberAvoidableMisclassificationEstimator misclassification_estimator
+
+        +void init_trigger(trigger_context)
+        +void inform_new_model(model_id, ...)
+
+        -float compute_regret_metric()
+    }
+
+
+    Trigger <|-- BatchedTrigger
+    BatchedTrigger <|-- CostTrigger
+    PerformanceTriggerMixin <|-- CostTrigger
+
+    CostTrigger *-- "1" CostTracker
+    CostTrigger *-- "1" IncorporationLatencyTracker
+    CostTrigger <|-- DataIncorporationLatencyCostTrigger
+    CostTrigger <|-- AvoidableMisclassificationCostTrigger
+
+    DataIncorporationLatencyCostTrigger *-- "1" DataIncorporationLatencyCostTriggerConfig
+    AvoidableMisclassificationCostTrigger *-- "1" AvoidableMisclassificationCostTriggerConfig
+
+    style PerformanceTriggerMixin fill:#DDDDDD,stroke:#A9A9A9,stroke-width:2px
+
+    style DataIncorporationLatencyCostTrigger fill:#C5DEB8,stroke:#A9A9A9,stroke-width:2px
+    style AvoidableMisclassificationCostTrigger fill:#C5DEB8,stroke:#A9A9A9,stroke-width:2px
+```
+
+### `CostTrigger` Hierarchy
+
+Both `DataIncorporationLatencyCostTrigger` and `AvoidableMisclassificationCostTrigger` track the cost of triggering and convert a regret metric (e.g., data incorporation latency or avoidable misclassification latency) into the training time unit with a user-defined conversion factor.
```

Comment:
while data incorporation latency is intuitive (and explained below), I don't have an intuitive understanding of avoidable misclassification latency

Reply:
Conceptually, it's not that different from data incorporation latency. It's simply the time that misclassified samples stayed unaddressed through retraining. 

I would even argue that it makes more sense than data incorporation latency as misclassifications are eventually the metric that should incentivize retraining, so we don't want them to stay "unaddressed" for too long. The regular number of samples is rather unhelpful. For a stationary dataset, the model's predictive capabilities could stay constant while data incorporation latency cumulates up, leading to another **unnecessary** trigger. But let's see what the experiments will result in.